test/e2e/deployment/deployment.go

```diff
@@ -0,0 +1,63 @@
+package deployment
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+// Deployment offers the configuration options of OVN-Kubernetes to e2e test cases
+type Deployment interface {
+	OVNKubernetesNamespace() string
+	ExternalBridgeName() string
+}
+
+var deployment Deployment
+
+func Set() {
+	// upstream currently uses KinD as its preferred platform infra, so if we detect KinD, its upstream
+	if isKind() {
+		deployment = newUpstream()
+	}
+	if isOpenShift() {
+		deployment = newOpenshift()
+	}
+	if deployment == nil {
+		panic("failed to determine the deployment")
```

Comment:
"deployment" is kind of generic and also already in use for a k8s type... That might or might not be a problem in context, but for instance, this error message is pretty ambiguous and hard to figure out

Reply:
ya, renamed to deploymentconfig - i hope to remove this once ovn-k has a config api.